### PR TITLE
Document SSM parameters used in deployment

### LINKED ISSUE #174: Documentation - Document SSM parameters used in operation
## Type of Issue:

- [x] Enhancement: New feature or request

#### Issue Creator Checklist

- [x] This issue has been thoroughly documented below; a developer should be able to understand the issue by reading it.

## Background

This project has a fairly consistent approach to supplying the deploy/build with envirnoment specific information.
This pattern is typically referred to as our "SSM pattern".  #102 introduced this pattern; please see it for details.

In short, however, our "SSM pattern" utilizes [AWS Systems Manager Parameter Store ](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) to hold environment specific information, both secret and non secret.

The serverless framework code will fetch information it needs from the serverless.yml

Example: https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/services/app-api/serverless.yml#L32
In this example, we are setting the custom variable infrastructure type to one of the following, in this order:
- The value of /configuration/<current branch name>/infrastructure/type, if that parameter exists in SSM
- The value of /configuration/default/infrastructure/type, if that parameter exists in SSM
- Else, "development"

With this pattern, we can set a default (at /configuration/default/.....) to apply to all branches, but maintain the ability to override the default with /configuration/<current branch name/....

We use this pattern and this hierarchy in many places in the quickstart.  This issue is to document what SSM variables are used by the deployment.

## Proposed Approach

The most important aspect of this issue is to collect the existing possible SSM configurations.  The simplest option is to do this manually... go into each service, see what SSM params are in use, and document them somewhere.
An automated approach to collecting SSM params seems better, but is not a requirement here.

It's thought that having each service independently list its SSM options (maybe in the serverless.yml, or maybe in a README.md inside each service) is good.

Listing them all together in the top level README might also be good.

These are just ideas.

## Value

Consumers of the quickstart will have a list of SSM parameters used by the deployment, and should have an easier experience forking and setting up their project.

## AC:

- All SSM parameters used by the deployment are documented in a sensible fashion.

### PR DESCRIPTION
**Pull request by joeaero19**
## Purpose
This PR is to document all ssm parameters that the various services uses.
_Describe the problem or feature in addition to a link to the issues._

#### Linked Issues to Close
closes #174 

## Approach
The SSM parameters, both required and optional, were documented in README.md files in each service's directory.  An explanation of our general SSM strategy/pattern was also added in each README.

A comment was added to each appropriate serverless.yml file that instructs the reader to look at the README for configuration parameters and usage.

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/app-api/serverless.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## set the values for the following ssm parameters in AWS SSM Parameter store for app-api service. You will need to set the values and also a default value to fall to if the required value is not present.
+# 1. iamPath:  /configuration/${self:custom.stage}/iam/path    &  /configuration/default/iam/path
```
Comment (dwhitecl) [inline services/app-api/serverless.yml:2]: @joeaero19  a few things
can you document what the purpose of each ssm paramater is ... the "what" it is and the "why" we're using it. 
Another thing is we should explain what self:custom.stage is ... I know what it is , I know the pattern in which we use that by specifying the branch name but that may not be obvious to a consumer of the quickstart. We should explain that pattern

Comment (dwhitecl) [reply services/app-api/serverless.yml:2]: also we should note which ssm param is required or optional

File: services/app-api/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# app-api
+
+## Configuration - AWS Systems Manager Parameter Store (SSM)
+
+The following values are used to configure the deployment of this service (see below for more background and context).
+| Parameter | Required? | Accepts a default? | Accepts a branch override? | Purpose |
+| --- | :---: | :---: | :---: | --- |
+| .../iam/path | N | Y | Y | Specifies the [IAM Path](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-friendly-names) at which all IAM objects should be created. The default value is "/". nb:The path variable in IAM is used for grouping related users and groups in a unique namespace, usually for organizational purposes.|
```
Comment (mdial89f) [inline services/app-api/README.md:8]: what's nb: ?

File: services/app-api/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+| .../iam/permissionsBoundaryPolicy | N | Y | Y | Specifies the [IAM Permissions Boundary](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) that should be attached to all IAM objects. nb: A permissions boundary is an advanced feature for using a managed policy to set the maximum permissions that an identity-based policy can grant to an IAM entity.|
```
Comment (mdial89f) [inline services/app-api/README.md:9]: what's nb: ?

File: services/ui/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+| Parameter | Required? | Accepts a default? | Accepts a branch override? | Purpose |
+| --- | :---: | :---: | :---: | --- |
+| .../iam/path | N | Y | Y | Specifies the [IAM Path](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-friendly-names) at which all IAM objects should be created. The default value is "/". nb:The path variable in IAM is used for grouping related users and groups in a unique namespace, usually for organizational purposes.|
+| .../iam/permissionsBoundaryPolicy | N | Y | Y | Specifies the [IAM Permissions Boundary](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) that should be attached to all IAM objects. nb: A permissions boundary is an advanced feature for using a managed policy to set the maximum permissions that an identity-based policy can grant to an IAM entity.|
+| route53HostedZoneId | N | Y | Y | This is the Id of the Amazon route53 hosted zone.|
+| cloudfrontCertificateArn | N | Y | Y | The arn for the clodfront distribution certificate.|
+| cloudfrontDomainName | N | Y | Y | The domain name for the cloudfront distribution.|
```
Comment (mdial89f) [inline services/ui/README.md:12]: cloudfrontDomainName does not accept a default value.  Any domain name specified is intrinsically bound to that specific environment.

File: services/ui/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+| route53DomainName | N | Y | Y | Route 53 domain name for the site if applicable.|
```
Comment (mdial89f) [inline services/ui/README.md:13]: route53DomainName also does not accept a default value.

File: services/ui-auth/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+| Parameter | Required? | Accepts a default? | Accepts a branch override? | Purpose |
+| --- | :---: | :---: | :---: | --- |
+| .../iam/path | N | Y | Y | Specifies the [IAM Path](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-friendly-names) at which all IAM objects should be created. The default value is "/". nb:The path variable in IAM is used for grouping related users and groups in a unique namespace, usually for organizational purposes.|
+| .../iam/permissionsBoundaryPolicy | N | Y | Y | Specifies the [IAM Permissions Boundary](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) that should be attached to all IAM objects. nb: A permissions boundary is an advanced feature for using a managed policy to set the maximum permissions that an identity-based policy can grant to an IAM entity.|
+| sesSourceEmailAddress | N | Y | Y | The email address with which the apllication sends the email. This email address must be verified in SES.|
+| okta_metadata_url | N | Y | Y | The SAML Metadata url for Okta, gotten from Okta's admin console from the app.|
```
Comment (mdial89f) [inline services/ui-auth/README.md:11]: okta_metadata_url does not accept a default value.  Only a branch specific value can be specified, just like cloudfront/route53 domain names.

**Review by mdial89f — CHANGES_REQUESTED** [review]
A few variables are marked as "accepts default: Y" but should be N.
I see "nb:" in a bunch of files... I marked two of those files, but stopped because I think it's just copy/paste in all.  I don't know what nb: is but just remove from all if its an accident.

Let me know, thanks, looks good otherwise.

File: services/app-api/README.md (hunk 2 above)
Comment (joeaero19) [reply services/app-api/README.md:8]: sorry nb means note or pay attention.

Comment (joeaero19) [reply services/app-api/README.md:8]: I am deleting that from the file.

File: services/ui-auth/README.md (hunk 6 above)
Comment (joeaero19) [reply services/ui-auth/README.md]: i have corrected this.

File: services/ui/README.md (hunk 5 above)
Comment (joeaero19) [reply services/ui/README.md]: I have corrected this

**Review by mdial89f — APPROVED** [review]

**Pull request merged**